spec/mongoid/traversable_spec.rb

```diff
@@ -251,4 +251,328 @@
+      end
+
+      it "is changed in the child: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello")
+      end
+    end
+
+    context "when the discriminator key is changed in the parent" do
+      before do
+        Instrument.discriminator_key = "hello2"
+      end
+
+      after do
+        Instrument.discriminator_key = nil
+      end
+
+      it "doesn't change the global setting" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+
+      it "changes in the parent" do
+        expect(Instrument.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Piano" do
+        expect(Piano.discriminator_key).to eq("hello2")
+      end
+
+      it "changes in the child class: Guitar" do
+        expect(Guitar.discriminator_key).to eq("hello2")
+      end
+
+      context 'when discriminator key is set to nil in parent' do
+        before do
+          Instrument.discriminator_key = nil
+        end
+
+        it "doesn't change the global setting" do
+          expect(Mongoid.discriminator_key).to eq("_type")
+        end
+
+        it 'uses global setting' do
+          expect(Instrument.discriminator_key).to eq("_type")
+        end
+
+        it "changes in the child class: Piano" do
+          expect(Piano.discriminator_key).to eq("_type")
+        end
+  
+        it "changes in the child class: Guitar" do
+          expect(Guitar.discriminator_key).to eq("_type")
+        end
+      end
+
+      context "when resetting the discriminator key after nil" do
+        before do
+          Instrument.discriminator_key = nil
+          Instrument.discriminator_key = "hello4"
+        end
+
+        it "doesn't change the global setting" do
+          expect(Mongoid.discriminator_key).to eq("_type")
+        end
+
+        it 'has the correct value' do
+          expect(Instrument.discriminator_key).to eq("hello4")
+        end
+
+        it "changes in the child class: Piano" do
+          expect(Piano.discriminator_key).to eq("hello4")
+        end
+  
+        it "changes in the child class: Guitar" do
+          expect(Guitar.discriminator_key).to eq("hello4")
+        end
+      end
+    end
+
+    context "when the discriminator key is changed in the child" do
+      let(:guitar) do 
+        Guitar.discriminator_key = "hello3"
+      end
+
+      it "raises an error" do
+        expect do
+          guitar
+        end.to raise_error(Mongoid::Errors::InvalidDiscriminatorKeyTarget)
+      end
+
+      it "doesn't change in that class" do
+        expect(Guitar.discriminator_key).to eq("_type")
+      end
+
+      it "doesn't change the global setting" do
+        expect(Mongoid.discriminator_key).to eq("_type")
+      end
+
+      it "doesn't change in the sibling" do
+        expect(Piano.discriminator_key).to eq("_type")
+      end
+
+      it "doesn't change in the parent" do
+        expect(Instrument.discriminator_key).to eq("_type")
+      end
```

Comment:
For these tests, I would add a line before the expectation that invokes guitar and throws away the exception
```
guitar rescue nil
```
While the first test within this context invokes guitar, I think you'd want to be able to run any of these tests individually and have it still do everything you want it to do.